test/integration/test_utils.mjs

```diff
@@ -581,6 +581,22 @@ async function kbFocusPrevious(page) {
   await awaitPromise(handle);
 }
 
+async function switchToEditor(name, page, disable = false) {
+  const modeChangedHandle = await createPromise(page, resolve => {
+    window.PDFViewerApplication.eventBus.on(
+      "annotationeditormodechanged",
+      resolve
+    );
```

Comment:
This listener is never being removed, which probably isn't what you want. However, fixing this just requires adding one line here:
```suggestion
    window.PDFViewerApplication.eventBus.on(
      "annotationeditormodechanged",
      resolve,
      { once: true }
    );
```